anvil/src/eth/backend/executor.rs

```diff
@@ -258,11 +267,31 @@ impl<'a, 'b, DB: Db + ?Sized, Validator: TransactionValidator> Iterator
 
         trace!(target: "backend", "[{:?}] executing", transaction.hash());
         // transact and commit the transaction
-        let ExecutionResult { exit_reason, out, gas_used, logs, .. } =
-            evm.inspect_commit(&mut inspector);
+        let exec_result = match evm.inspect_commit(&mut inspector) {
+            Ok(exec_result) => exec_result,
+            Err(err) => {
+                warn!(target: "backend", "[{:?}] failed to execute: {:?}", transaction.hash(), err);
+                return Some(TransactionExecutionOutcome::DatabaseError(
+                    transaction,
+                    DatabaseError::TransactionNotFound(H256::from_str("0x").unwrap()),
+                ))
+            }
+        };
         inspector.print_logs();
 
-        if exit_reason == Return::OutOfGas {
+        let (exit_reason, gas_used, out, logs) = match exec_result {
+            ExecutionResult::Success { reason, gas_used, logs, output, .. } => {
+                (eval_to_instruction_result(reason), gas_used, Some(output), Some(logs))
+            }
+            ExecutionResult::Revert { gas_used, .. } => {
+                (InstructionResult::Revert, gas_used, None, None)
+            }
+            ExecutionResult::Halt { reason, gas_used } => {
+                (halt_to_instruction_result(reason), gas_used, None, None)
```

Comment:
Unsure if the gas refund calculation is accurate here (and on the other places this `ExecutionResult` is matched)